autoevents.py

```diff
@@ -138,13 +151,76 @@ def _convert_time(self, time_string, local=True):
+            datestring = eventchange_datetime_UTC.strftime("%Y-%m-%d %H:%M:%S")
+            sql_args = (
+                datestring,
+                datestring
+            )
+        dbreturn = self._mad['db_wrapper'].execute(sql_query, args=sql_args, commit=True)
+        self._mad['logger'].info(f'Event Watcher: Pokemon deleted by SQL query: {sql_query} arguments: {sql_args} return: {dbreturn}')
+
+    def _check_pokemon_resets(self):
+        #get current time to check for event start and event end
+        now = datetime.now()
+        self._mad['logger'].info("Event Watcher: Check Pokemon reset")
+        # check, if one of the pokemon event is just started or ended
+        for event in self._pokemon_events:
+            # event start during last check?
+            if self._last_pokemon_reset_check < event["start"] <= now:
+                self._mad['logger'].success(f'Event Watcher: Reset Pokemon (event start detected for event type: {event["type"]})')
+                # remove pokemon from MAD DB, which are scanned before event start and needs to be rescanned, adapt time from local to UTC time
+                self._reset_pokemon(event["start"] - timedelta(hours=self.tz_offset))
+                break
+            # event end during last check?
+            if self._last_pokemon_reset_check < event["end"] <= now:
+                self._mad['logger'].success(f'Event Watcher: Reset Pokemon (event end detected for event type: {event["type"]})')
+                # remove pokemon from MAD DB, which are scanned before event end and needs to be rescanned, adapt time from local to UTC time
+                self._reset_pokemon(event["end"] - timedelta(hours=self.tz_offset))
+                break
+        self._last_pokemon_reset_check = now;
+
+    def _reset_all_quests(self):
+        sql_query = "TRUNCATE trs_quest"
+        dbreturn = self._mad['db_wrapper'].execute(sql_query, commit=True)
+        self._mad['logger'].info(f'Event Watcher: Quests deleted by SQL query: {sql_query} return: {dbreturn}') 
+
     def _check_quest_resets(self):
+        #get current time to check for event start and event end
         now = datetime.now()
+        self._mad['logger'].info("Event Watcher: Check Quest reset")
+        # check, if one of the pokemon event is just started or ended
+        for event in self._quest_events:
+            # event starts during last check?
+            if "start" in self.__quests_reset_types.get(event["type"], []):
+                if self._last_quest_reset_check < event["start"] <= now:
+                    self._mad['logger'].success(f'Event Watcher: Reset Quests (event start detected for event type: {event["type"]})')
+                    # remove all quests from MAD DB
+                    self._reset_all_quests()
+                    self._mad["mapping_manager"].update()
+                    break
+            # event end during last check?
+            if "end" in self.__quests_reset_types.get(event["type"], []):
```

Comment:
code duplication:

please change the the if statement to something like this:

```py
to_check = self.__quests_reset_types.get(event["type"], [])
if "start" in to_check or "end" in to_check:
```

Reply:
corrected by similar code as you used in quest reschedule function. See https://github.com/ccev/mp-eventwatcher/pull/8/commits/eb4d253da60e39a7864ca7ba52bb7bab3860ff67